pkg/dispatcher/cache/event_handlers.go

```diff
@@ -210,6 +228,39 @@ func (dc *DispatcherCache) setResourceBinding(rb *workv1alpha2.ResourceBinding)
 		return
 	}
 
+	// Check if this ResourceBinding is associated with a DataSourceClaim (DSC).
```

Comment:
I think this part of the logic should not be coupled in `setResourceBinding`, which will affect the existing functionality, especially when the users have not installed ds/dsc crds. If these codes are userd as a dispatcher session plugin, this plugin adds an `Allocatable` callback, which only checks whether the placement annotation of rb is added. If not, this plugin will not allow rb pass, is this OK?